core/src/main/java/org/springframework/security/oauth2/server/authorization/InMemoryOAuth2AuthorizationService.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.server.authorization;
+
+import org.springframework.security.oauth2.core.OAuth2AccessToken;
+import org.springframework.util.Assert;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * In-memory implementation of {@link OAuth2AuthorizationService}.
+ *
+ * @author Krisztian Toth
+ */
+public class InMemoryOAuth2AuthorizationService implements OAuth2AuthorizationService {
+	private final Map<TokenWithType, OAuth2Authorization> tokenAuthorizationMap = new ConcurrentHashMap<>();
```

Comment:
The `TokenWithType` key will limit the search capabilities. What if we need to provide a finder for `principalName`, eg. find all authorizations for a specific user.

Let's change this to a `List` and use `list.stream().filter()` in `findByTokenAndTokenType()`

Reply:
I agree, though the implementation requirements stated that `Map` should be used. Updated in 
fd7ab91